File: SearchView(java)/app/src/main/java/coder/mtk/newappjava/viewmodel/NewsViewModel.java
package coder.mtk.newappjava.viewmodel;

import android.util.Log;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import coder.mtk.newappjava.api.ApiService;
import coder.mtk.newappjava.model.News;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class NewsViewModel extends ViewModel {

    MutableLiveData<News> result = new MutableLiveData<>();
    MutableLiveData<Boolean> loadError = new MutableLiveData<>();
    MutableLiveData<Boolean> loading = new MutableLiveData<>();

    public MutableLiveData<News> getResult() {
        return result;
    }

    public MutableLiveData<Boolean> getLoadError() {
        return loadError;
    }

    public MutableLiveData<Boolean> getLoading() {
        return loading;
    }

    public void loadResult (){
        loading.setValue(true);
        String apiKey = "8f4b1cee839b43ab80645baee4a33881";

        Call<News> newsCall = ApiService.getNews("us","Technology",apiKey);
        newsCall.enqueue(
                new Callback<News>() {
                    @Override
                    public void onResponse(Call<News> call, Response<News> response) {
                        Log.d("responseresult>>",response.body().toString());
                        if (response.isSuccessful() && response.body()!= null){
                            loading.setValue(false);
                            result.setValue(response.body());

                        }
                    }

                    @Override
                    public void onFailure(Call<News> call, Throwable t) {
                        loading.setValue(false);
                        loadError.setValue(true);
                        Log.d("Response>>>>",t.toString());
                    }
                }
        );
    }

}

File: NewsApi/app/src/main/java/coder/mtk/newsapi/viewmodel/NewsViewModel.kt
package coder.mtk.newsapi.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import coder.mtk.newsapi.api.NewsApiClient
import coder.mtk.newsapi.model.News
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class NewsViewModel : ViewModel(){
    private var result : MutableLiveData<News> = MutableLiveData()
    private var errorMessage : MutableLiveData<String> = MutableLiveData()
    private var progressBar : MutableLiveData<Boolean> = MutableLiveData()
    private var errorStatus: MutableLiveData<Boolean> = MutableLiveData()
    fun  getResult() : LiveData<News> = result
    fun getErrorMessage () : LiveData<String> = errorMessage
    fun getErrorStatus () : LiveData<Boolean> = errorStatus
    fun getProgress () : LiveData<Boolean> = progressBar

    fun loadResult (){
        var apiClient = NewsApiClient()
        var apiCall = apiClient.getTopHeadLine()

        apiCall.enqueue(
            object : Callback<News>{
                override fun onFailure(call: Call   <News>, t: Throwable) {
                    errorMessage.value = t.toString()
                    progressBar.value = true
                    errorStatus.value=true
                }

                override fun onResponse(call: Call<News>, response: Response<News>) {
                    result.value = response.body()
                    progressBar.value =false
                    errorStatus.value = false
                }

            }
        )
    }
}